fix(ci): KafkaSqlStartupVerificationIT is flaky due to container startup timeouts

## Description

`KafkaSqlStartupVerificationIT` in the `utils/extra-tests` module is flaky on CI. It has caused
failures on at least two unrelated PRs (#8400, #8401) in the same day.

## Failure Modes Observed

### 1. Kafka container startup timeout (PR #8400)

The `StrimziKafkaCluster` fails to start within the default timeout during `@BeforeAll`,
preventing all 13 tests from running:

```
java.lang.RuntimeException: Timed out while starting Kafka containers
    at io.strimzi.test.container.StrimziKafkaCluster.start(StrimziKafkaCluster.java:785)
    at io.apicurio.registry.utils.tests.infra.KafkaInfra.start(KafkaInfra.java:52)
    at io.apicurio.registry.utils.tests.KafkaSqlStartupVerificationIT.beforeAll(KafkaSqlStartupVerificationIT.java:30)
Caused by: java.util.concurrent.TimeoutException
```

### 2. Registry container startup timeout (PR #8401)

Kafka starts successfully, but the registry container does not emit the expected
`KafkaSQL storage bootstrapped in ...` log message within the 75-second
`KAFKASQL_STARTUP_TIMEOUT`, causing `testGoodSnapshotsTopicConfig1` to fail:

```
org.opentest4j.AssertionFailedError:
Expecting value to be true but was false
    at ...KafkaSqlStartupVerificationIT.testGoodSnapshotsTopicConfig1(KafkaSqlStartupVerificationIT.java:90)
```

This second mode is harder to distinguish from a genuine failure because
`startRegistryContainer` catches `ContainerLaunchException` and returns `false` — the
same value the test expects for *invalid* topic configurations. When the timeout is the
cause, a "good config" test incorrectly sees `false`.

## Root Cause Analysis

1. **No startup timeout on the Kafka cluster itself.** `KafkaInfra.start()` calls
   `StrimziKafkaClusterBuilder.build()` and then `start()` without configuring
   `withStartupTimeout()`. The Strimzi container defaults may be too tight for
   resource-constrained CI runners.

2. **Registry container timeout is borderline.** `KAFKASQL_STARTUP_TIMEOUT` is 75 seconds.
   The comment says observed worst-case is ~49 seconds, but under heavy CI load (parallel
   jobs, shared runners) this margin can be insufficient. The test class runs 13 tests
   sequentially, each starting/stopping a registry container, which compounds resource
   pressure.

3. **Test ordering sensitivity.** `testGoodSnapshotsTopicConfig1` runs mid-suite after
   several start/stop cycles. Accumulated container overhead and resource contention from
   prior tests may push it past the timeout.

## Suggested Fixes

- **Increase the Kafka cluster startup timeout.** Configure a startup timeout on the
  `StrimziKafkaClusterBuilder` in `KafkaInfra.start()` (e.g., 120 seconds) to handle slow
  CI runners.
- **Increase `KAFKASQL_STARTUP_TIMEOUT`.** Raise from 75 seconds to 120 seconds to match
  the approach taken in #8389 for operator tests.
- **Add retry via `rerunFailingTestsCount`.** The `extra-tests` Maven profile should
  include the `failsafe.rerunFailingTestsCount` property (as done in #8388 for other
  modules) so transient container issues don't block the pipeline.

## References

- PR #8400: https://github.com/Apicurio/apicurio-registry/pull/8400
- PR #8401: https://github.com/Apicurio/apicurio-registry/pull/8401
- Related: #8389 (increase operator test timeouts)
- Related: #8388 (add rerunFailingTestsCount for flaky retry)

--- a/utils/extra-tests/src/test/java/io/apicurio/registry/utils/tests/infra/AbstractRegistryInfra.java
+++ b/utils/extra-tests/src/test/java/io/apicurio/registry/utils/tests/infra/AbstractRegistryInfra.java
@@ -11,15 +11,16 @@ import org.testcontainers.containers.wait.strategy.Wait;
 
 import java.time.Duration;
 import java.util.Map;
+import java.util.concurrent.TimeoutException;
 
 public abstract class AbstractRegistryInfra {
 
     private static final Logger log = LoggerFactory.getLogger(AbstractRegistryInfra.class);
 
     // KafkaSQL bootstrap involves sequential Kafka consumer polls at 5s intervals
-    // plus SQL initialization. 75s provides margin over observed ~49s worst case
-    // under CI load without masking genuinely hung containers.
-    private static final Duration KAFKASQL_STARTUP_TIMEOUT = Duration.ofSeconds(75);
+    // plus SQL initialization. Configurable via -Dtest.extra.timeout.kafkasql-startup=<seconds>.
+    private static final Duration KAFKASQL_STARTUP_TIMEOUT = Duration.ofSeconds(
+            Integer.getInteger("test.extra.timeout.kafkasql-startup", 120));
 
     private final String name;
 
@@ -50,6 +51,14 @@ public abstract class AbstractRegistryInfra {
             registryContainer.followOutput(new Slf4jLogConsumer(log).withPrefix(name));
             return true;
         } catch (ContainerLaunchException ex) {
+            if (ex.getCause() instanceof TimeoutException) {
+                throw new AssertionError(
+                        "Registry container '" + name + "' startup timed out after "
+                                + KAFKASQL_STARTUP_TIMEOUT.toSeconds() + "s"
+                                + " — this is an infrastructure issue, not a validation failure. "
+                                + "Override with -Dtest.extra.timeout.kafkasql-startup=<seconds>",
+                        ex);
+            }
             log.error("Error starting {} container: {}", name, ex.getMessage(), ex);
             return false;
         }

--- a/utils/extra-tests/src/test/java/io/apicurio/registry/utils/tests/infra/KafkaInfra.java
+++ b/utils/extra-tests/src/test/java/io/apicurio/registry/utils/tests/infra/KafkaInfra.java
@@ -49,7 +49,9 @@ public class KafkaInfra implements AutoCloseable {
                     .withNumberOfBrokers(1) // required
                     .withSharedNetwork()
                     .build();
+            long startTime = System.currentTimeMillis();
             kafkaCluster.start();
+            log.info("Kafka cluster started in {} ms", System.currentTimeMillis() - startTime);
         }
     }
 

